File: Structural/Decorator/JavaScript/sugarDecorator.js
// sugarDecorator.js

const CoffeeDecorator = require("./coffeeDecorator");

class SugarDecorator extends CoffeeDecorator {
  getDescription() {
    return this.decoratedCoffee.getDescription() + ", Sugar";
  }

  getCost() {
    return this.decoratedCoffee.getCost() + 0.5;
  }
}

module.exports = SugarDecorator;

File: Structural/Decorator/Python/sugar_decorator.py
# sugar_decorator.py

from coffee_decorator import CoffeeDecorator

class SugarDecorator(CoffeeDecorator):
  def get_description(self) -> str:
    return self._decorated_coffee.get_description() + ", Sugar"

  def get_cost(self) -> float:
    return self._decorated_coffee.get_cost() + 0.5
